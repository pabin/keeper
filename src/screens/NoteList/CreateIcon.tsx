import React from 'react';
import { StyleSheet, View } from 'react-native';
import Icon from 'react-native-vector-icons/dist/FontAwesome';

import { colors } from '../../styles/colors';

const CreateIcon = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <Icon
        onPress={() => navigation.navigate('CreateNotes')}
        name="plus"
        size={24}
        color={colors.white}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: 60,
    height: 60,
    bottom: 24,
    right: 24,
    zIndex: 3,
    elevation: 5,
    borderRadius: 30,
    alignItems: 'center',
    justifyContent: 'center',
    position: 'absolute',
    backgroundColor: colors.orange,
  },
});

export default CreateIcon;
