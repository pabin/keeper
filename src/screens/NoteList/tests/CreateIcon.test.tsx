import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react-native';

import CreateIcon from '../CreateIcon';

describe('CreateIcon', () => {
  afterEach(cleanup);

  const navigateMock = jest.fn();
  const navigation = {
    navigate: navigateMock,
  };

  test('renders correctly', () => {
    const { toJSON } = render(<CreateIcon navigation={navigation} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('navigates correctly', () => {
    const { getByTestId } = render(<CreateIcon navigation={navigation} />);

    const createNoteIcon = getByTestId('createNoteIcon');
    fireEvent.press(createNoteIcon);
    expect(navigateMock).toHaveBeenCalledWith('CreateNotes');
  });
});
