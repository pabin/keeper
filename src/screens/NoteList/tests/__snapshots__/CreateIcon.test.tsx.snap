// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateIcon renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#01579B",
      "borderRadius": 30,
      "bottom": 24,
      "elevation": 5,
      "height": 60,
      "justifyContent": "center",
      "position": "absolute",
      "right": 24,
      "width": 60,
      "zIndex": 3,
    }
  }
>
  <Text
    allowFontScaling={false}
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="createNoteIcon"
  >
    ÔÅß
  </Text>
</View>
`;
